### PR TITLE
feature: EthHashInfo add customAvatar fallback

### PR DESCRIPTION
**Pull request by nicosampler**
We need to handle the case for when `customAvatar` is broken.

- I added a new property called `customAvatarFallaback` that receives an URL to callback in case the main image is broken
- I no `customAvatarFallaback` was provided it fallback to Identicon.

These changes are needed for https://app.zenhub.com/workspaces/safe-multisig---web-5ce554debb310a35b2f8b6f8/issues/gnosis/safe-react/2133

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr123--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr123--safereactcomponents.review.gnosisdev.com

File: src/ethereum/EthHashInfo/index.tsx [hunk 1]
```diff
@@ -70,42 +71,62 @@ const EthHashInfo = ({
...
-    )}
+}: Props): React.ReactElement => {
+  const [fallbackToIdenticon, setFallbackToIdenticon] = useState(false);
+
+  const setAppImageFallback = (
+    error: SyntheticEvent<HTMLImageElement, Event>
+  ): void => {
+    if (customAvatarFallback && !fallbackToIdenticon) {
+      error.currentTarget.onerror = null;
+      error.currentTarget.src = customAvatarFallback;
```
Comment (katspaugh) [inline src/ethereum/EthHashInfo/index.tsx:87]: Maybe we could avoid manipulating DOM directly?
```
setFallbackSrc(customAvatarFallback);

...

<StyledImg
  src={fallbackSrc || customAvatar}
  ...
/>
```

File: src/ethereum/EthHashInfo/index.tsx [hunk 2]
```diff
@@ -54,6 +54,7 @@ type Props = {
   textSize?: ThemeTextSize;
   showAvatar?: boolean;
   customAvatar?: string;
+  customAvatarFallback?: string;
```
Comment (katspaugh) [inline src/ethereum/EthHashInfo/index.tsx:57]: Is this image always going to be the `<>` icon? If so, we could just store the SVG in the repo. Or at least use it as the default value.

**Review by katspaugh — CHANGES_REQUESTED** [review]

File: src/ethereum/EthHashInfo/index.tsx (hunk 1 above)
Comment (nicosampler) [reply src/ethereum/EthHashInfo/index.tsx:87]: you're right, I copied it directly from SRC.
what do you suggest? `ref` perhpahs?

File: src/ethereum/EthHashInfo/index.tsx (hunk 2 above)
Comment (nicosampler) [reply src/ethereum/EthHashInfo/index.tsx:57]: There are 2 different repositories. 
`<>` is needed at this moment in SRC, but as this lib is also helpful for SApp I would not use it as a default one.

File: src/ethereum/EthHashInfo/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/ethereum/EthHashInfo/index.tsx:87]: Just a state variable, see the code snippet in my previous comment.

**Review by katspaugh — APPROVED** [review]
Thanks! Please update the snapshots.

**Review by katspaugh — APPROVED** [review]

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr123--safereactcomponents.review.gnosisdev.com

**Pull request merged**